### PR TITLE
RavenDB-5377: TxInfo debug EP

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

File: src/Voron/Impl/LowLevelTransaction.cs [hunk 1]
```diff
@@ -206,6 +207,8 @@ private LowLevelTransaction(LowLevelTransaction previous)
 
         public LowLevelTransaction(StorageEnvironment env, long id, TransactionPersistentContext transactionPersistentContext, TransactionFlags flags, IFreeSpaceHandling freeSpaceHandling, ByteStringContext context = null)
         {
+            TxStartTime = DateTime.UtcNow;
+            ThreadId = Thread.CurrentThread.ManagedThreadId;
```
Comment (ayende) [inline src/Voron/Impl/LowLevelTransaction.cs]: We don't want this, this is actually quite slow to compute.

Comment (ayende) [reply src/Voron/Impl/LowLevelTransaction.cs]: Also, we don't typically remain on the same thread.

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Voron/Impl/LowLevelTransaction.cs (hunk 1 above)
Comment (aviviadi) [reply src/Voron/Impl/LowLevelTransaction.cs]: So which thread do we want to present ?

Comment (aviviadi) [reply src/Voron/Impl/LowLevelTransaction.cs]: and what else do we want to see in this report ?

Comment (aviviadi) [reply src/Voron/Impl/LowLevelTransaction.cs]: Ok.. I'll use NativeMemory.ThreadStats

File: src/Raven.Server/Documents/Handlers/Debugging/TransactionDebugHandler.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                context.Write(writer, ToJson(results));
+            }
+            return Task.CompletedTask;
+        }
+
+        private DynamicJsonArray ToJson(List<TransactionInfo> txInfos)
+        {
+            if (txInfos.Count < 1)
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Debugging/TransactionDebugHandler.cs]: Why, same result if you remove this line

File: src/Voron/Impl/LowLevelTransaction.cs (hunk 1 above)
Comment (ayende) [reply src/Voron/Impl/LowLevelTransaction.cs]: Waiting for this change

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Voron/Impl/LowLevelTransaction.cs [hunk 3]
```diff
@@ -145,6 +146,8 @@ private LowLevelTransaction(LowLevelTransaction previous)
             env.Options.AssertNoCatastrophicFailure();
 
             FlushInProgressLockTaken = previous.FlushInProgressLockTaken;
+            CurrentTransactionHolder = previous.CurrentTransactionHolder;
+            TxStartTime = previous.TxStartTime;
```
Comment (ayende) [inline src/Voron/Impl/LowLevelTransaction.cs:150]: Should we copy the previous start time?

Comment (aviviadi) [reply src/Voron/Impl/LowLevelTransaction.cs:150]: no...

Comment (ayende) [reply src/Voron/Impl/LowLevelTransaction.cs:150]: what is your reasoning?

Comment (aviviadi) [reply src/Voron/Impl/LowLevelTransaction.cs:150]: https://github.com/ravendb/ravendb/pull/2185

**Pull request merged**